### PR TITLE
Add brokerDeleteInactiveTopicsEnabled parameter

### PR DESCRIPTION
**Pull request by vroyer**
Add a configurable brokerDeleteInactiveTopicsEnabled .

### CODE REVIEW

File: helm-chart-sources/pulsar/templates/broker-deployment/broker-configmap.yaml [hunk 1]
```diff
@@ -58,6 +58,9 @@ data:
     {{- end }}
     {{- end }}
   clusterName: {{ template "pulsar.fullname" . }}
+  {{- if .Values.broker.brokerDeleteInactiveTopicsEnabled }}
+  brokerDeleteInactiveTopicsEnabled: {{ .Values.brokerDeleteInactiveTopicsEnabled }}
```
Comment (cdbartholomew) [inline helm-chart-sources/pulsar/templates/broker-deployment/broker-configmap.yaml:62]: This doesn't match the if above. `.Values.broker.brokerDeleteInactiveTopicsEnabled` vs `.Values.brokerDeleteInactiveTopicsEnabled` I think having the setting under `broker` is the right place for it.

**Review by cdbartholomew — CHANGES_REQUESTED** [review]
Can we add the new setting with a default value to the `values.yaml` file? That way users will now it's available.

File: helm-chart-sources/pulsar/templates/broker-deployment/broker-configmap.yaml (hunk 1 above)
Comment (vroyer) [reply helm-chart-sources/pulsar/templates/broker-deployment/broker-configmap.yaml]: Fix the broker config map and add a default in the values.yaml.

**Comment by lhotari** [comment]
It's already possible to set configuration values in `broker.configData`. Anything sub-keys will be passed as environment variables to brokers.

There are a lot of different broker configuration options that could be configured.
Does it make sense to start adding another layer for passing configuration keys to the broker?

Example of adding  `brokerDeleteInactiveTopicsEnabled` setting to `broker.configData` in `dev-values.yaml`:
```
broker:
  component: broker
  replicaCount: 1
  ledger:
    defaultEnsembleSize: 1
    defaultAckQuorum: 1
    defaultWriteQuorum: 1
  resources:
    requests:
      memory: 600Mi
      cpu: 0.3
  configData:
    PULSAR_MEM: "\"-Xms400m -Xmx400m -XX:MaxDirectMemorySize=200m -XX:+ExitOnOutOfMemoryError\""
    brokerDeleteInactiveTopicsEnabled: "false"
```

**Comment by eolivelli** [comment]
this is an old patch.
At that time I told to @vroyer about configData and he was unblocked

I believe we can close this patch

is it correct @vroyer  ?

**Comment by vroyer** [comment]
Yes, I agree, the configData can do the job (but it was undocumented for the newbie I was when I needed to set this parameter).

**Comment by eolivelli** [comment]
@vroyer @lhotari  can you close this PR ? I do not have permissions

**Comment by lhotari** [comment]
Ok, I'll close this PR.

**Pull request closed without merging**